components/nimbus/src/persistence.rs

```diff
@@ -442,6 +597,729 @@ mod tests {
+            json!({
+                "slug": "feature-id-empty",
+                "status":
+                    {
+                        "Enrolled":
+                            {
+                                "enrollment_id": "801ee64b-0b1b-44a7-be47-5f1b5c189086",
+                                "reason": "Qualified",
+                                "branch": "control",
+                                "feature_id": ""
+                            }
+                        }
+            }),
+        ]
+    }
+
+    #[test]
+    /// Migrating v1 to v2 involves finding enrollments that
+    /// don't contain all the feature stuff they should and discarding.
+    fn test_migrate_v1_to_v2_enrollment_discarding() -> Result<()> {
+        let _ = env_logger::try_init();
+        let tmp_dir = TempDir::new("migrate_v1_to_v2")?;
+
+        let rkv = Database::open_rkv(&tmp_dir)?;
+        let meta_store = SingleStore::new(rkv.open_single("meta", StoreOptions::create())?);
+        let enrollment_store =
+            SingleStore::new(rkv.open_single("experiments", StoreOptions::create())?);
+        let mut writer = rkv.write()?;
+
+        meta_store.put(&mut writer, "db_version", &1)?;
+
+        // write invalid enrollments
+        let invalid_feature_enrollments = &get_invalid_feature_enrollments();
+        assert_eq!(2, invalid_feature_enrollments.len());
+
+        for enrollment in invalid_feature_enrollments {
+            //log::debug!("enrollment = {:?}", enrollment);
+            enrollment_store.put(
+                &mut writer,
+                enrollment["slug"].as_str().unwrap(),
+                enrollment,
+            )?;
+        }
+
+        writer.commit()?;
+
+        let db = Database::new(&tmp_dir)?;
+
+        // The enrollments with invalid feature_ids should have been discarded
+        // during migration; leaving us with none.
+        let enrollments = db
+            .collect_all::<ExperimentEnrollment>(StoreId::Enrollments)
+            .unwrap();
+        //log::debug!("enrollments = {:?}", enrollments);
+
+        assert_eq!(enrollments.len(), 0);
+
+        Ok(())
+    }
+
+    /// Migrating v1 to v2 involves finding experiments that
+    /// don't contain all the feature stuff they should and discarding.
+    #[test]
+    fn test_migrate_v1_to_v2_experiment_discarding() -> Result<()> {
+        let _ = env_logger::try_init();
+        let tmp_dir = TempDir::new("migrate_v1_to_v2_enrollment_discarding")?;
+
+        let rkv = Database::open_rkv(&tmp_dir)?;
+        let meta_store = SingleStore::new(rkv.open_single("meta", StoreOptions::create())?);
+        let experiment_store =
+            SingleStore::new(rkv.open_single("experiments", StoreOptions::create())?);
+        let mut writer = rkv.write()?;
+
+        meta_store.put(&mut writer, "db_version", &1)?;
+
+        // write a bunch of invalid experiments
+        let invalid_feature_experiments = &get_invalid_feature_experiments();
```

Comment:
I'm thinking I'll go with v1_db_schema_experiments, since it's otherwise easy to confuse the db schema with the DTO schema (which would maybe ideally be the same thing, but I'm concerned that coupling them too tightly is not the thing to take an now, if we want to do it at all).

Reply:
(The 1 has already been chosen as the existing DB_VERSION)

Reply:
Went with `db_v1_experiments_with_missing_feature_fields`.  The version numbers have already been set in them migration code since before this patch.

Reply:
Fixed.